src/core.js

```diff
@@ -155,37 +151,37 @@ function unblockAndAdvanceQueue() {
 }
 
 export function begin() {
-	var i, l,
-		modulesLog = [];
 
-	// If the test run hasn't officially begun yet
-	if ( !config.started ) {
+	if ( config.started ) {
+		unblockAndAdvanceQueue();
+		return;
+	}
 
-		// Record the time of the test run's beginning
-		config.started = now();
+	// The test run hasn't officially begun yet
+	// Record the time of the test run's beginning
+	config.started = now();
 
-		// Delete the loose unnamed module if unused.
-		if ( config.modules[ 0 ].name === "" && config.modules[ 0 ].tests.length === 0 ) {
-			config.modules.shift();
-		}
-
-		// Avoid unnecessary information by not logging modules' test environments
-		for ( i = 0, l = config.modules.length; i < l; i++ ) {
-			modulesLog.push( {
-				name: config.modules[ i ].name,
-				tests: config.modules[ i ].tests
-			} );
-		}
+	// Delete the loose unnamed module if unused.
+	if ( config.modules[ 0 ].name === "" && config.modules[ 0 ].tests.length === 0 ) {
+		config.modules.shift();
+	}
 
-		// The test run is officially beginning now
-		emit( "runStart", globalSuite.start( true ) );
-		runLoggingCallbacks( "begin", {
-			totalTests: Test.count,
-			modules: modulesLog
-		} ).then( unblockAndAdvanceQueue );
-	} else {
-		unblockAndAdvanceQueue();
+	// Avoid unnecessary information by not logging modules' test environments
+	const l = config.modules.length;
+	var modulesLog = [];
```

Comment:
Feel free to convert individual functions from var to const/let in their entirety.